lib/remote.provider.js

```diff
@@ -1,7 +1,113 @@
 'use strict';
 
+const jstp = require('@metarhia/jstp');
+
 const { StorageProvider } = require('./provider');
+const { RemoteCursor } = require('./remote.cursor');
+
+class RemoteProvider extends StorageProvider {
+  constructor() {
+    super();
+    this.connection = null;
+  }
+
+  // Open StorageProvider
+  //    options - options connection jstp
+  //    callback - function(err, RemoteProvider)
+  open(options, callback) {
+    jstp[options.transport].connect(
+      ...options.connectionArgs,
+      (error, connection) => {
+        if (error) {
+          callback(error);
+          return;
+        }
+        this.connection = connection;
+        callback(null, this);
+      }
+    );
+  }
+
+  // Close RemoteProvider
+  //   callback - function(err)
+  close(callback) {
+    if (!this.connection) {
+      callback();
+      return;
+    }
+
+    this.connection.close(callback);
+  }
+
+  // Get record from GlobalStorage
+  //   id - globally unique record id
+  //   callback - function(err, obj)
+  get(id, callback) {
+    this.connection.callMethod('provider', 'get', [id], callback);
+  }
+
+  // Set record in GlobalStorage
+  //   record - record to be stored
+  //   callback - function(err)
+  set(record, callback) {
+    this.connection.callMethod('provider', 'set', [record], callback);
+  }
+
+  // Create record in GlobalStorage
+  //   category - category of record
+  //   record - record to be stored
+  //   callback - function(err, id)
+  create(category, record, callback) {
+    this.connection.callMethod(
+      'provider',
+      'create',
+      [category, record],
+      callback
+    );
+  }
+
+  // Update record in GlobalStorage
+  //   category - category of record
+  //   query - record, example: { Id }
+  //   patch - record, fields to update
+  //   callback - function(err, count)
+  update(category, query, patch, callback) {
+    this.connection.callMethod(
+      'provider',
+      'update',
+      [category, query, patch],
+      callback
+    );
+  }
+
+  // Delete record in GlobalStorage
+  //   category - category of record
+  //   query - record, example: { Id }
+  //   callback - function(err, count)
+  delete(category, query, callback) {
+    this.connection.callMethod(
+      'provider',
+      'delete',
+      [category, query],
+      callback
+    );
+  }
 
-class RemoteProvider extends StorageProvider {}
+  // Select record from GlobalStorage
+  //   category - category of record
+  //   query - fields conditions
+  // Returns: Cursor
+  select(category, query) {
+    this.connection.callMethod(
+      'provider',
+      'select',
+      [category, query],
+      cursorId => {
+        const cursor = new RemoteCursor(cursorId, this.connection);
+        return cursor;
```

Comment:
`select` will return undefined because you attempt to return a value from a callback. You should probably pass connection and arguments to `RemoteConstructor` and make a jstp call there.

Reply:
BTW where is `RemoteCursor` implementation?

Reply:
As of https://github.com/metarhia/globalstorage/commit/71fe5bdc81e2edfb5f7aceeacd2eb3cb4c2e4aaa, select will also fetch all needed data, so you should call it only when the user tries to fetch the data (there is no `fetch` method).